src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,99 @@
+
+  const {
+    id,
+    title,
+    userId,
+    completed,
+  } = todo;
+
+  const fetchUser = useCallback(async () => {
+    setIsLoading(true);
+
+    try {
+      const fetchedUser = await getUser(userId);
+
+      setUser(fetchedUser);
+    } catch (e) {
+      if (e instanceof Error) {
+        alert(e.message);
+      } else {
+        alert('Unexpected error');
+      }
+    } finally {
+      setIsLoading(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    fetchUser();
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
+      {isLoading
+        ? <Loader />
+        : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${id}`}
+              </div>
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={onClose}
+              />
+            </header>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+            <div className="modal-card-body">
+              <p className="block" data-cy="modal-title">
+                {title}
+              </p>
+
+              <p className="block" data-cy="modal-user">
+                <strong className={
+                  classNames({
+                    'has-text-success': completed,
+                    'has-text-danger': !completed,
+                  })
+                }
+                >
+                  {completed ? 'Done' : 'Planned'}
+                </strong>
+
+                {' by '}
+                {user && (
+                  <a href={`mailto:${user?.email}`}>
```

Comment:
you're already checked that user exists before rendering this element, do you need to check it one more time inside?